### PR TITLE
[scaffolder] Include entity metadata in ctx.templateInfo 

### PR DESCRIPTION
**Pull request by dagda1**
## Hey, I just made a Pull Request!

Implementation of #13223.

An `entityMetadata` field has been added to the `TemplateInfo` type that contains the `Template` entity's metadata.

```ts
export type TemplateInfo = {
  entityRef: string;
  baseUrl?: string;

  /**
   * The Template entity's metadata
   */
  entityMetadata: EntityMeta;
};
```

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder-common/api-report.md [hunk 1]
```diff
@@ -64,5 +65,6 @@ export const templateEntityV1beta3Validator: KindValidator;
 export type TemplateInfo = {
   entityRef: string;
   baseUrl?: string;
+  entityMetadata: EntityMeta;
```
Comment (benjdlambert) [inline plugins/scaffolder-common/api-report.md:68]: I think that this should be optional. There's a chance that new code could pick up old tasks which don't have it set. So let's make it optional for now to enforce runtime checking of the existence for now.

Maybe also let's have it on something like this shape?

```ts
export type TemplateInfo = {
  entityRef: string;
  entity?: {
    metadata?: EntityMeta;
  }
}
```

Comment (benjdlambert) [reply plugins/scaffolder-common/api-report.md:68]: This way it's similar to the `${{ user.entity.metadata }}` helper too.

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Looking good, just a few things with the API

File: plugins/scaffolder-common/api-report.md (hunk 1 above)
Comment (dagda1) [reply plugins/scaffolder-common/api-report.md]: @benjdlambert thank you for the feedback, and I have implemented the changes.

**Review by benjdlambert — APPROVED** [review]
Nice! Thanks! :pray:

**Pull request merged**